CONTRIBUTING.md

```diff
@@ -203,6 +203,15 @@ cd packages/opentelemetry-module-name
 npm run watch
 ```
 
+#### TypeScript version & update policy
+
+TypeScript version used to compile the pacakges is `v5.0.4`. If you plan to use any of the packages from this
+repository to make your own application or package instrumentation it is recommended to use same version or higher.
+
+<!-- Ref: https://github.com/open-telemetry/opentelemetry-js/pull/5145#issuecomment-2518263890 -->
+As update policy OpenTelemetry JS will follow DefinitelyType's [support policy for TypeScript](https://github.com/DefinitelyTyped/DefinitelyTyped#support-window)
+which sets a support window of 2 years.
+
```

Comment:
It may be good to have this be more visible by also stating something similar in `README.md` (`README_NEXT.md`) - while it's important for contributors to know how we're using typescript, there's an effect on end-users that we should communicate.

Reply:
resolving this and re-adding the comment on the current re-based version.